elfp.c

```diff
@@ -84,15 +189,32 @@ test_one(char *f, Elf *elf, int flags)
     if (gelf_getehdr(elf, &ehdr) == NULL)
 	return;
 
+    if ((flags & P_BUILDID)) {
+	size_t sz;
+	uint8_t *data = get_buildid(elf, &sz);
+
+	if (data) {
+	    b = alloca(sz * 2 + 1);
+	    data2hex(data, sz, b);
+	}
+    }
+
     if ((flags & P_REL) && (ehdr.e_type == ET_REL))
 	goto out_print;
 
     if ((flags & P_EXEC) && (ehdr.e_type == ET_EXEC))
 	goto out_print;
 
+    if ((flags & P_DEBUG) && has_debuginfo(elf))
+	goto out_print;
+
     /* arguably should print if P_OTHER, but, nah. */
-    if (ehdr.e_type != ET_DYN)
+    if (ehdr.e_type != ET_DYN) {
+	if ((flags & P_BUILDID) && b)
+	    goto out_print;
+
 	return;
```

Comment:
So, my intent there was to limit buildid /the same way/ everything else is limited *unless* it's the only selector.  But that actually needs another conditional later like:
```diff
diff --git a/elfp.c b/elfp.c
index f89e0f2..86db255 100644
--- a/elfp.c
+++ b/elfp.c
@@ -224,6 +224,9 @@ test_one(const char *f, Elf *elf, int flags)
        goto out_print;
     }
 
+    if (((flags & ~P_NEWLINE)  == P_BUILDID) && b)
+       goto out_print;
+
     return;
 
 out_print:
```